src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -124,17 +124,25 @@ uct_rc_mlx5_iface_poll_tx(uct_rc_mlx5_iface_common_t *iface)
     ucs_trace_poll("rc_mlx5 iface %p tx_cqe: ep %p qpn 0x%x hw_ci %d", iface, ep,
                    qp_num, hw_ci);
 
+    /* Process completions before updating resources to avoid OOO.
+     * Some resources (RDMA_READ credits) are updated in operation completion
+     * callbacks, so need to call all relevant callbacks prior dispatching
+     * arbiter. In this case ordering is preserved, because:
+     * - If nothing is sent from the completion callback, we just update
+     *   resources and dispatch arbiter.
+     * - If something needs to be sent from completion callback, but not all
+     *   needed resources available, it will be added to the pending queue
+     *   (and handled in dispatch below in the proper order).
+     */
+    uct_rc_mlx5_txqp_process_tx_cqe(&ep->super.txqp, cqe, hw_ci);
```

Comment:
I'd wrap this function and comment to the separate function in order to avoid comment duplication
DCI related item from the comment could go as:
```
- For DC-only: DCI is released after handling completion callbacks ...
```

Otherwise, we should maintain two comments